[[ Bug 22980 ]] Initialize uninitialized MCPlatformMenuItemHighlight variable

livecodepanos opened the pull request:
This patch ensures a `MCPlatformMenuItemHighlight` variable is initialized. Previously this was not the case, and this probably resulted in menuitems having a '-' at the beginning when LC was running in an ARM Mac.

runrevmark reviewed (changes requested):
This needs a report in the QC and a corresponding release note.

Update the commit title to be [[ Bug XXXX ]] ... as usual.

runrevmark commented on docs/notes/bugfix-22980.md:
```diff
@@ -0,0 +1 @@
+# Initialize uninitialized variable in menu item code
```
So the release note should be:
```
# Menu items which don't have a mark now display correctly on Big Sur when running on ARM processors
```

livecodepanos committed:
[[ Bug 22980 ]] Initialize uninitialized MCPlatformMenuItemHighlight variable

This patch ensures a MCPlatformMenuItemHighlight variable is initialized.
Previously this was not the case, and this probably resulted in menuitems
displayed incorrectly on Big Sur when running on ARM processors.

runrevmark commented:
@livecode-vulcan review ok 0a2c3eb

livecode-vulcan commented:
💙 review by @runrevmark ok 0a2c3eb8a4088017febf27d5ebfcb6d0cce17df6

livecode-vulcan commented:
😎 test success 0a2c3eb8a4088017febf27d5ebfcb6d0cce17df6

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/927)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/335)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/334)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/333)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/334)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2452)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/30)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/105)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/28)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/101)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2479)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2468)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2466)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2503)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2439)
